### PR TITLE
Creation of HPA template

### PR DESCRIPTION
**Pull request by Vidya2606**
# Description

Create new HPA template

## Type of change
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: template/manifests/PodDisruptionBudget/manifest/draft.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    type: "string"
+    kind: "resourceName"
+    description: "the name of the application"
+    default:
+      value: "frontend"
+  - name: "PARTOF"
+    type: "string"
+    kind: "part-of"
+    description: "the label to identify which project the resource belong to"
+    default:
```
Comment (bfoley13) [inline template/manifests/PodDisruptionBudget/manifest/draft.yaml:15]: same here, no default value

File: template/manifests/PodDisruptionBudget/manifest/hpa.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{.APPNAME}}
+  minReplicas: {{.MINIMUMREPLICAS}}
+  maxReplicas: {{.MAXIMUMREPLICAS}}
+  metrics:
+    - type: Resource
+      resource:
+        name: {{.CPU}}
```
Comment (bfoley13) [inline template/manifests/PodDisruptionBudget/manifest/hpa.yaml:19]: This has to be the resource name, `memory` or `cpu`

File: template/manifests/PodDisruptionBudget/manifest/draft.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    kind: "resource"
+    description: "Specifies that the HPA will monitor memory for scaling (in Mi)"
+    default:
+      value: 100
+  - name: "CPU"
+    type: "int"
+    kind: "resource"
+    description: "Specifies that the HPA will monitor CPU utilization for scaling (in m)"
+    default:
+      value: 100
```
Comment (bfoley13) [inline template/manifests/PodDisruptionBudget/manifest/draft.yaml:46]: these two should just be combined into a string variable `RESOURCETYPE` that can be `cpu` or `memory`

File: template/manifests/PodDisruptionBudget/manifest/draft.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      value: "online-store"
+  - name: "GENERATORLABEL"
+    type: "string"
+    kind: "label"
+    description: "the label to identify who generated the resource"
+    default:
+      value: "draft"
+  - name: "MINIMUMREPLICAS"
+    type: "int"
+    kind: "replicas"
```
Comment (bfoley13) [inline template/manifests/PodDisruptionBudget/manifest/draft.yaml:25]: I think `replicaCount` would be a better kind definition for this and maxreplicas

**Review by Tatsinnit — APPROVED** [review]
👍❤️ looks good to me. Thank you for kind ping!!

File: template/manifests/PodDisruptionBudget/manifest/draft.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  - name: "RESOURCETYPE"
+    type: "string"
+    kind: "scalingResourceType"
+    description: "Specifies that the HPA will monitor cpu or memory for scaling"
+    default:
+      value: "cpu"
+  - name: "AVGUTILIZATION"
+    type: "int"
+    kind: "autoscaling"
+    description: "the HPA will scale up if the CPU utilization exceeds the default value"
```
Comment (OliverMKing) [inline template/manifests/PodDisruptionBudget/manifest/draft.yaml:40]: this description is incorrect if they change the resource type

File: template/manifests/PodDisruptionBudget/manifest/draft.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+description: "This template is used to create a PodDisruptionBudget for an application"
+type: "podDisruptionBudget"
+variables:
+  - name: "APPNAME"
+    type: "string"
+    kind: "resourceName"
+    description: "the name of the application"
+  - name: "PARTOF"
+    type: "string"
+    kind: "part-of"
```
Comment (bosesuneha) [inline template/manifests/PodDisruptionBudget/manifest/draft.yaml:11]: nit: can you make this camelCase to maintain uniformity

Comment (Vidya2606) [reply template/manifests/PodDisruptionBudget/manifest/draft.yaml:11]: it is part of the label: app.kubernetes.io/part-of: online-store, so I kept it as is!

Comment (bosesuneha) [reply template/manifests/PodDisruptionBudget/manifest/draft.yaml:11]: Makes sense

**Review by bosesuneha — COMMENTED** [review]
lgtm

**Review by bosesuneha — APPROVED** [review]

File: template/manifests/PodDisruptionBudget/manifest/draft.yaml [hunk 7]
```diff
@@ -0,0 +1,42 @@
+templateName: "podDisruptionBudget-manifest"
+description: "This template is used to create a PodDisruptionBudget for an application"
+type: "podDisruptionBudget"
+variables:
+  - name: "APPNAME"
+    type: "string"
+    kind: "resourceName"
```
Comment (bfoley13) [inline template/manifests/PodDisruptionBudget/manifest/draft.yaml:7]: the `kind` params should follow the other draft.yaml definitions:
https://github.com/Azure/draft/blob/d45a77ac45e34416de33bd4579d733ce53cfd8e8/template/deployments/manifests/draft.yaml#L16C12-L16C34

File: template/manifests/PodDisruptionBudget/manifest/draft.yaml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+      value: 5
+  - name: "RESOURCETYPE"
+    type: "string"
+    kind: "scalingResourceType"
+    description: "specifies the resource type (e.g., cpu or memory) to be monitored for scaling"
+    default:
+      value: "cpu"
+  - name: "AVGUTILIZATION"
+    type: "int"
+    kind: "autoscaling"
```
Comment (bfoley13) [inline template/manifests/PodDisruptionBudget/manifest/draft.yaml:39]: make this kind `scalingResourceUtilization`

File: template/manifests/PodDisruptionBudget/manifest/draft.yaml [hunk 9]
```diff
@@ -0,0 +1,42 @@
+templateName: "podDisruptionBudget-manifest"
+description: "This template is used to create a PodDisruptionBudget for an application"
+type: "podDisruptionBudget"
```
Comment (bfoley13) [inline template/manifests/PodDisruptionBudget/manifest/draft.yaml:3]: this should be `manifest`

**Review by bfoley13 — APPROVED** [review]

**Pull request merged**